packages/ui/src/common/components/Search/SearchResultsModal.tsx

```diff
@@ -19,16 +19,21 @@ import { ModalWithDataCall } from '@/common/providers/modal/types'
 import { ThreadItemBreadcrumbs } from '@/forum/components/threads/ThreadItemBreadcrumbs'
 import { ForumRoutes } from '@/forum/constant'
 
+import { useDebounce } from '../../hooks/useDebounce'
+
 export type SearchResultsModalCall = ModalWithDataCall<'SearchResults', { search: string }>
 
 export const SearchResultsModal = () => {
   const { hideModal, modalData } = useModal<SearchResultsModalCall>()
-
   const [search, setSearch] = useState(modalData.search)
   const [activeTab, setActiveTab] = useState<SearchKind>('FORUM')
   const { forum, forumPostCount, isLoading } = useSearch(search, activeTab)
-  const pattern = useMemo(() => (search ? RegExp(escapeStringRegexp(search), 'ig') : null), [search])
-
+  const isValid = () => !debouncedSearch || debouncedSearch.length === 0 || debouncedSearch.length > 2
+  const debouncedSearch = useDebounce(search, 400)
+  const pattern = useMemo(
+    () => (isValid() && search ? RegExp(escapeStringRegexp(search), 'ig') : null),
+    [debouncedSearch]
+  )
```

Comment:
Looks like this doesn't work on the preview branch: https://dao-git-fork-oleksanderkorn-3470-joystream.vercel.app/#/forum
I'm still able to search for 2 chars on the side pane (although it takes time since it's a big query)

@oleksanderkorn could you debug this please ?

Reply:
I put the search limit back to 3 in `useSearch` should be ok now.